source/dynamic_menu.sqf

```diff
@@ -0,0 +1,160 @@
+/*
+    Author: UltraNoobian (Brenden Cai)
+
+    Description:
+    Dynamically generate and update the Support menus, then display the root menu.
+
+    Parameter(s):
+    None.
+
+    Usage:
+    [] call "dynamic_menu.sqf";
+
+    Returns:
+    - Nil -
+*/
+menuItemFormatter = {
+    format [
+        "[""%1"", [%2], """", -5, [[""expression"", ""%3""]], ""1"", ""%4""]",
+        _this select 0,
+        _this select 1,
+        _this select 2,
+        _this select 3
+    ];
+};
+
+_itemsCount = 0;
+_offensiveItems = [];
+_logisticsItems = [];
+_transportItems = [];
+
+// Critical note: If you use escape quotes as part of your params to menuItemFormatter,
+// You must double-escape those quotes
+// eg. "Hello ""mate""" -> "Hello """"mate"""""
+// eg. 'Hello "mate"' -> "Hello ""mate""'
+
+
+//Offensive Submenu checks
+if(support_mortar_available) then {
+    _text = ["Mortar (2 CP)", 2, '_null = [8, 50, 5, 3, 2, 2, ""grenade""] execVM ""support\mapclickarty.sqf""', 1] call menuItemFormatter;
```

Comment:
Instead of execVM'ing everytime (It recompiles and preprocesses the script everytime). You should compile it once and store it in a variable. And then `spawn` or `call` it.